src/main/java/ch/wisv/areafiftylan/products/service/TicketServiceImpl.java

```diff
@@ -198,6 +202,15 @@ public void cancelTicketTransfer(String token) {
         return ticketRepository.findByPickupService_True();
     }
 
+    @Override
+    public Ticket assignTicketToUser(Long ticketId, String username) {
+        Ticket ticket = getTicketById(ticketId);
+        User user = userService.getUserByUsername(username);
+        ticket.setOwner(user);
+        return ticketRepository.save(ticket);
```

Comment:
That is correct, it is not flushed to the database yet, but when failure occurs, data will not be saved in the database yet and lost. If you read [this](http://stackoverflow.com/questions/21203875/difference-between-save-and-saveandflush-in-spring-data-jpa) stackoverflow answer, the changes will only be written to the database until the flush or commit methods are called. I would think the saveAndFlush method would fit better in our current usage of all the methods. We can change later to a save and commit/flush structure.